lib/Driver/Compilation.cpp

```diff
@@ -762,17 +765,19 @@ namespace driver {
       }
 
       assert(!Partition.empty());
-      maybeShuffleBatchable(Batchable);
-
-      size_t i = 0;
+      auto PartitionIndex = assignJobsToPartitions(Partition.size(),
+                                                   Batchable.size());
+      assert(PartitionIndex.size() == Batchable.size());
       auto const &TC = Comp.getToolChain();
-      for (const Job *Cmd : Batchable) {
-        maybeAdvanceToNextPartition(i, Partition, Batchable.size());
-        std::vector<const Job*> &P = Partition[i];
+      for (size_t i = 0; i < Batchable.size(); ++i) {
```

Comment:
If you want, there's an `llvm::for_each` that can do parallel iteration over two containers.

Reply:
Assuming you mean swift::for_each (in include/swift/Basic/STLExtras.h), nice!

Reply:
Oops, forgot it was us and not them.